### ISSUE TITLE
Conv2D failing with a 'dilation_rate' KeyError

### ISSUE THREAD

**Issue by mortendahl** [tf-encrypted/tf-encrypted#624]
This can be reproduced by running the PySyft TFE notebooks.

**Comment by mortendahl** [comment]
Issue seems to be interference between `default_args_check` in TFE Keras and the Keras hooking in PySyft, which erases the signature of layers:

```
UnknownLayerArgError: Argument 'dilation_rate' is not part of the signature for 'Conv2D' layers: odict_keys(['self', 'args', 'kwargs'])
```

**Comment by mortendahl** [comment]
@jvmancuso thought you might have some ideas!

**Comment by jvmancuso** [comment]
if anyone from PySyft is looking here, a temporary workaround would be to downgrade your install of tf-encrypted to 0.5.6 via `pip install -U tf-encrypted==0.5.6`. we'll close this issue when a fix is merged to master, and then leave another comment when that fix makes it into the next release.

**Comment by jvmancuso** [comment]
This ended up being far easier to fix on the PySyft side by properly wrapping the decorated function with `functools.wraps`.  This issue can be closed once https://github.com/OpenMined/PySyft/pull/2371 is merged.
